doc-deploy-stable/action.yml

```
@@ -378,28 +359,34 @@ runs:
       with:
         level: "INFO"
         message: >
-          Generate the redirection link. Since this is the stable documentation
-          deployment action, there is always a 'version/stable' folder. Hence,
-          redirection link always points to this folder.
+          Create an index.html page copied from index.html in the stable version.
+          Update all the href and source links to point to the stable or dev version.
 
-    - name: "Generate the redirection URL"
+    - name: "Use the latest 'index.html' in base index.html page"
       shell: bash
       run: |
-        echo "REDIRECTION_URL=https://${{ inputs.cname }}/version/stable/" >> $GITHUB_ENV
+        if [[ -f 'version/stable/index.html' ]]; then
+          cp version/stable/index.html index.html
+        else
+          cp version/dev/index.html index.html
+        fi
 
-    - name: "Create the 'index.html' redirection file"
+    - name: "Update all href and source links to version/stable"
       shell: bash
       run: |
-        echo "<!DOCTYPE html>" > index.html
-        echo "<meta charset=\"utf-8\">" >> index.html
-        if [ -z "${{ inputs.redirection-title }}" ]; then
-          echo "<title>Redirecting to ${{ env.REDIRECTION_URL }}</title>" >> index.html
+        if [[ ! -f 'index.html' ]]; then
+          echo "Error: The 'index.html' file does not exist." >&2
+          exit 1
+        fi
+        # Determine the version directory based on the existence of stable index.html
+        if [[ -f 'version/stable/index.html' ]]; then
+          version_dir="version/stable"
         else
-          echo "<title>${{ inputs.redirection-title }}</title>" >> index.html
+          version_dir="version/dev"
         fi
-        echo "<meta name=\"description\" content=\"${{ inputs.redirection-description }}\">" >> index.html
-        echo "<meta http-equiv=\"refresh\" content=\"0; URL=${{ env.REDIRECTION_URL }}\">" >> index.html
-        echo "<link rel=\"canonical\" href=\"${{ env.REDIRECTION_URL }}\">" >> index.html
+        # Update URLs
+        sed -i "s#href=\"\([^h]\)#href=\"$version_dir/\1#g" index.html
+        sed -i "s#src=\"\([^h]\)#src=\"$version_dir/\1#g" index.html
```

Sorry for the confusion. Rather than checking references starting with `h`, let's check for the complete protocols. Also, using here `|` as splitter for the `sed` command:

```suggestion
        sed -i 's|href="\(\(http\|https\)\://\)\{0,1\}|href="version/stable/|g' index.html
        sed -i 's|src="\(\(http\|https\)\://\)\{0,1\}|src="version/stable/|g' index.html
```